File: acoustics_toolbox_r2020/Python/Bellhop/Tests/DW/Arctic/arcticB_gb.py
#==================================================================
#  
#  Bellhop: Arctic sound speed profile
#  Mexilhoeira Grande, Sex Jul 28 16:49:10 WEST 2017
#  Written by Tordar
#  
#==================================================================

# ipython: run arcticB_gb

from os import *
import sys
from numpy import *
from scipy.io import *
from pylab import *
sys.path.append ("/home/orodrig/FORdoc/at/Python/")
from readshd import *

rs = 0.0

print("Bellhop - Arctic profile:")
print("Coherent Acoustic Pressure calculation") 
print("Geometric gaussian beams")

system("bellhop.exe arcticB_gb")

filename = 'arcticB_gb.shd'
xs = nan
ys = nan
pressure,geometry = readshd(filename,xs,ys)

zs     = geometry["zs"]
rarray = geometry["rarray"]; rarraykm = rarray/1000
zarray = geometry["zarray"]

Dmax = zarray[-1]
rmax = rarray[-1]; rmaxkm = rmax/1000

p = squeeze( pressure, axis=(0,1) )
tl = -20*log10( abs( p ) )

tl[tl > 100.0] = 100.0

figure(1)
imshow(tl,extent=[0,rmaxkm,0,Dmax],aspect='auto',cmap='jet_r',origin='lower',vmin=40,vmax=90)
cb = colorbar()
cb.ax.invert_yaxis()
plot(rs,-zs,marker="<",markersize=16,color="k")
xlabel('Range (km)')
ylabel('Depth (m)')
title('Bellhop - Arctic waveguide')
ylim(Dmax,0)

show()

print("done.")

File: acoustics_toolbox_r2020/Python/Bellhop/Tests/DW/Arctic/arcticB_gb.m
clear all, close all

addpath /home/orodrig/FORdoc/at/Matlab/ReadWrite

disp('Bellhop - Arctic profile') 

system('bellhop.exe arcticB_gb');

[ PlotTitle, PlotType, freq, atten, Pos, p ] = read_shd( 'arcticB_gb.shd' );

zs = Pos.s.depth;

rarray = Pos.r.range; rarraykm = rarray/1000;
zarray = Pos.r.depth;

rmax = max( rarray ); rmaxkm = rmax/1000;
zmax = max( zarray );

pressure = squeeze( p );

tl = -20*log10( abs( pressure ) ); 

tej = flipud( jet( 256 ) );

figure(1), hold on
plot(0,zs,'ko',0,zs,'m*','MarkerSize',16) 
pcolor(rarraykm,zarray,tl), shading interp, colormap( tej ), caxis([40 90]), colorbar
axis([0 rmaxkm 0 zmax])
hold off
view(0,-90)
xlabel('Range (km)')
ylabel('Depth (m)')
title('Bellhop - Arctic profile, 200 Hz')

disp('done.')
